### PR TITLE
IGNITE-20341 SQL Calcite: Upgrade calcite to 1.34.0

### PR DESCRIPTION
**Pull request by ivandasch**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/RexImpTable.java [hunk 1]
```diff
@@ -782,11 +786,41 @@ public static Expression multiplyDivide(Expression e, BigDecimal multiplier,
...
+            }
+        }
+    }
+
     /** Implementor for the {@code TRIM} function. */
     private static class TrimImplementor extends AbstractRexCallImplementor {
         /** */
         TrimImplementor() {
-            super(NullPolicy.STRICT, false);
+            super("trim", NullPolicy.STRICT, false);
```
Comment (alex-plekhanov) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/RexImpTable.java:823]: Remove `getVariableName()` method

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/RexImpTable.java [hunk 2]
```diff
@@ -376,6 +380,10 @@ public class RexImpTable {
         defineMethod(UNIX_MICROS, "unixMicros", NullPolicy.STRICT);
         defineMethod(DATE_FROM_UNIX_DATE, "dateFromUnixDate", NullPolicy.STRICT);
         defineMethod(UNIX_DATE, "unixDate", NullPolicy.STRICT);
+        defineMethod(DATE, "date", NullPolicy.STRICT);
```
Comment (alex-plekhanov) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/RexImpTable.java:383]: We override this call by `map.put(DATE, new CastImplementor());` later in this method, so looks like we don't need this call at all (if everithing works).

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java [hunk 3]
```diff
@@ -208,11 +225,67 @@ public RelDataType convert(SqlDataTypeSpec typeSpec) {
      * @return Validated node, its validated type and type's origins.
      */
     public ValidationResult validateAndGetTypeMetadata(SqlNode sqlNode) {
+        List<SqlNode> selectItems = null;
+        List<SqlNode> selectItemsNoStar = null;
+        SqlNode sqlNode0 = sqlNode instanceof SqlOrderBy ? ((SqlOrderBy)sqlNode).query : sqlNode;
```
Comment (alex-plekhanov) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java:230]: As far as I know UNION also can be as root node. 
Are you sure we can skip validation for nodes like OrderBy? Perhaps this block should be applied after `SqlNode validatedNode = validator().validate(sqlNode);` on `validatedNode`.

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/MetadataIntegrationTest.java [hunk 4]
```diff
@@ -41,11 +41,32 @@ public void trimColumnNames() {
...
+                .check();
+
+        assertQuery("select p.*, salary / 2, a.* from person p, address a where p.id = a.person_id")
+                .columnNames("ID", "NAME", "SALARY", "SALARY / 2", "ID", "PERSON_ID", "DATA")
+                .check();
 
         assertQuery("select count(_key), _key from person group by _key")
-            .columnNames("COUNT(_KEY)", "_KEY")
-            .check();
+                .columnNames("COUNT(_KEY)", "_KEY")
```
Comment (alex-plekhanov) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/MetadataIntegrationTest.java:68]: Redundant indent (for previous queries too).

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/MetadataIntegrationTest.java [hunk 5]
```diff
@@ -41,11 +41,32 @@ public void trimColumnNames() {
     /** */
     @Test
     public void columnNames() {
-        createAndPopulateTable();
+        sql("CREATE TABLE person (ID INT PRIMARY KEY, NAME VARCHAR, SALARY DOUBLE)");
+        sql("CREATE TABLE address (ID INT PRIMARY KEY, PERSON_ID INT, DATA VARCHAR)");
+
+        assertQuery("select salary , *, name, * from person")
```
Comment (alex-plekhanov) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/MetadataIntegrationTest.java:47]: Lets add cases for ORDER BY and UNION

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java (hunk 3 above)
Comment (ivandasch) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java:230]: Actually, we do not skip validation. This is used just to fix aliases later, after validation

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/RexImpTable.java (hunk 2 above)
Comment (ivandasch) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/RexImpTable.java:383]: Seems that `map.put(DATE, new CastImplementor());` isn't needed at all

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java [hunk 6]
```diff
@@ -268,10 +272,12 @@ public ValidationResult validateAndGetTypeMetadata(SqlNode sqlNode) {
...
                             SqlNode noStarItem = selectItemsNoStar.get(cnt);
 
+                            // The validator can transform SqlIdentifier to AS call. We should check whether
+                            // AS call is a real one and take the second operand from original one.
                             if (noStarItem instanceof SqlBasicCall
-                                    && ((SqlBasicCall)node).operandCount() == 2 && node.getKind() == SqlKind.AS
-                                    && ((SqlBasicCall)node).operand(0) instanceof SqlIdentifier
-                                    && ((SqlBasicCall)node).operand(1) instanceof SqlIdentifier) {
+                                    && isAsCall((SqlBasicCall)noStarItem)
+                                    && node instanceof SqlBasicCall
```
Comment (alex-plekhanov) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java:279]: It's already checked earlier (we are in `if (node instanceof SqlBasicCall)` branch). Also, this check can be moved to `isAsCall`.

**Review by alex-plekhanov — APPROVED** [review]

**Pull request merged**